Codeinwp/neve
# chore: proper error handling for incompatibility notice

### Summary
- Added proper error handling of the notice in case the update breaks
- Added proper permissions handling if the user cannot update themes or plugins
- The license notice is now hidden when the user cannot manage options
- Pro available updates aren't taken into account if the license is invalid
- Fixes issue where Theme Name wasn't passed properly through the white label filters


### Will affect visual aspect of the product

NO

### Test instructions

Prerequisites: 
- For the notice to appear you have to either have an older version of the plugin or the theme (**not both at the same time**).
- You can achieve that by altering the file headers (either in `style.css` for the theme, or in the `neve-pro-addon.php` file for the plugin.

Test Procedure (what has to work):
- User roles that are not capable of updating themes / plugins should not be shown the notice
- User roles that cannot manage options shouldn't be able to see the license notice or any subsequent notices that they are not allowed to take action on
- User roles that have proper capabilities should have the notices shown:
  - Users that can update plugins can see the notice when the plugin is outdated
  - Users that can update themes can see the notice when the theme is outdated
  - Users that can manage options can see the license notice
- The incompatibility notice should now work properly with the white-label module from the pro version. Before they were shown the actual name of the theme, instead of the whitelabeled one. 

I know this might be a bit hard to follow, so I've pinpointed the landmarks in the code of the PR. You can see them individually below, or take a look at the whole thing [here](https://github.com/Codeinwp/neve/pull/1662/files/76a9d84dbf2ebf17af2377e1c52e65f11c3372ab).


Closes Codeinwp/neve-pro-addon#723.
